sql/log.cc

```diff
@@ -8703,7 +8821,16 @@ MYSQL_BIN_LOG::queue_for_group_commit(group_commit_entry *orig_entry)
 bool
 MYSQL_BIN_LOG::write_transaction_to_binlog_events(group_commit_entry *entry)
 {
-  int is_leader= queue_for_group_commit(entry);
+  int is_leader;
+
+  if (binlog_free_flush.should_free_flush(entry) &&
```

Comment:
There's an invocation of  `trx_group_commit_leader()` inside which I fear not least 'cos
that new decision to compute the leader skips `queue_for_group_commit()` (and something else I would add later).
Would it be actually possible to execute all the renaming logics in the flush loop of `trx_group_commit_leader()` ? I [am pointing](https://github.com/MariaDB/server/pull/3402/commits/75e36f431bd08fe4e2b192b496456751466a018b#r1734958967) to the exact place ...

Reply:
moved the code to where you suggested.